ui/src/App.tsx

```diff
@@ -550,15 +598,18 @@ export function SHLinks() {
           ))}
         </React.Fragment>
       ))}
-
       <h4>
-        Custom Data Set
-        <button>Create new link</button>
+        New Custom Data Set
+        <button onClick={() => navigate('/health-links/new?custom=true', {state: {custom: true}})}>Create new dataset and link</button>
```

Comment:
This line is duplicating the `custom: true` as a URL param and also in a "state" object.

Are there reasons not to just use `custom=true` as a URL param? Then you could do something similar to what we have in place for the dataset ID (`let datasetId = Number(searchParams.get('ds'));`) today.

Reply:
I think we can just use `custom=true` as a URL param - I don't have much experience with URL params in React but after testing without the "state" object it still seems to work.

I pushed the rest of the changes as well, so this PR is ready for a re-review!